### PR TITLE
[Examples ios swift] Add CollectionView example app target

### PR DESCRIPTION
**Pull request by mrackwitz**
Adds a sample using a `UICollectionView`, which needs little different glue code to propagate changes from our fine-grained notifications to the collection view. This is mainly intended as a reference for our users.

/c @icanzilb @TimOliver

### CODE REVIEW

**Comment by mrackwitz** [comment]
CI failures are unrelated.

**Comment by mrackwitz** [comment]
ping and /c @jpsim

Completely reviewing this PR with all the boilerplate stuff in here is difficult. I'd appreciate a brief look at [CollectionViewController.swift](https://github.com/realm/realm-cocoa/blob/mr/example/collectionview/examples/ios/swift-2.2/CollectionView/CollectionViewController.swift).

File: examples/ios/swift-2.2/CollectionView/CollectionViewController.swift [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        label.text = object.title
+        dateLabel.text = Cell.dateFormatter.stringFromDate(object.date)
+    }
+}
+
+class CollectionViewController: UICollectionViewController {
+    var notificationToken: NotificationToken? = nil
+
+    lazy var realm = try! Realm()
+    lazy var results: Results<DemoObject> = {
```
Comment (jpsim) [inline examples/ios/swift-2.2/CollectionView/CollectionViewController.swift]: why is the closure useful here?

Comment (jpsim) [inline examples/ios/swift-2.2/CollectionView/CollectionViewController.swift]: wouldn't you just write this?

```swift
lazy var results = self.realm.objects(DemoObject)
```

**Comment by jpsim** [comment]
I don't get the motivation for this example. It seems to be exactly like the TableViewExample, except for the very few places where UICollectionView's API differs.

**Comment by mrackwitz** [comment]
These very few differences are exactly the motivation. Giving those of our users, who are not familiar with the API to propagate batched updates to a `UICollectionView`, a copy-paste ready boilerplate on how to achieve that with Realm.

**Comment by jpsim** [comment]
In my opinion, I don't think this provides enough differentiated value compared to our existing examples to include here. Closing.

**Pull request closed without merging**